recipes/setup_chef_server.rb

```diff
@@ -43,9 +43,9 @@
 # converge and complete the install.
 machine chef_server_hostname do
   if node['delivery-cluster']['chef-server']['existing']
-    recipe 'chef-server-12::delivery_setup'
+    run_list node['delivery-cluster']['chef-server']['run_list_existing']
   else
-    recipe 'chef-server-12'
+    run_list node['delivery-cluster']['chef-server']['run_list_new']
```

Comment:
I understand why you need the attributes.. I like that. Just put back `recipe` instead of `run_list` because we need them.